### PR TITLE
Fix text typos

### PR DESCRIPTION
**Pull request by huacnlee**
https://github.com/huacnlee/book-tw/actions/runs/3935914263/jobs/6732002327

### CODE REVIEW

File: src/appendix-08-terminology.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ instance               | 實體                          |
 iterative              | 疊代                          | 參考：[維基百科][iterative]
 iterator               | 疊代器                        | 參考：[維基百科][iterator]
 immutable              | 不可變                        | 參考：[維基百科][immutable]
-inheritance            | 繼承　                        |
```
Comment (huacnlee) [inline src/appendix-08-terminology.md:71]: Here have an invisible whitespace

File: src/appendix-03-derivable-traits.md [hunk 2]
```diff
@@ -87,7 +87,7 @@
 [creating-instances-from-other-instances-with-struct-update-syntax]:
 ch05-01-defining-structs.html#使用結構體更新語法從其他結構體建立實例
 [stack-only-data-copy]:
-ch04-01-what-is-ownership.html#只在堆疊上的資料拷貝copy
+ch04-01-what-is-ownership.html#只在堆疊上的資料拷貝 copy
```
Comment (wusyong) [inline src/appendix-03-derivable-traits.md]: This will become invalid markdown link. Is there a rule to exclude it?

**Comment by huacnlee** [comment]
@wusyong I have updated

File: src/ch10-02-traits.md [hunk 3]
```diff
@@ -48,7 +48,7 @@
 ```
 
 此程式碼會印出「1 則新推文：horse_ebooks: of course, as you probably already
-know, people」。
+know, people].
```
Comment (weihanglo) [inline src/ch10-02-traits.md]: Is it a false-positive?

File: src/ch19-03-advanced-traits.md [hunk 4]
```diff
@@ -269,6 +269,6 @@ Rust 並沒有限制不同特徵之間不能有同名的方法，也沒有阻止
 [newtype]: ch19-03-advanced-traits.html#使用新型別模式替外部型別實作外部特徵
 [對型別實作特徵]: ch10-02-traits.html#為型別實作特徵
 [iterator-特徵與-next-方法]: ch13-02-iterators.html#iterator-特徵與-next-方法
-[特徵-定義共享行為]: ch10-02-traits.html#特徵-定義共享行為
+[特徵 - 定義共享行為]: ch10-02-traits.html#特徵-定義共享行為
```
Comment (weihanglo) [inline src/ch19-03-advanced-traits.md]: Shouldn't correct this. The link will be broken.

File: src/ch20-01-single-threaded.md [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 我們會先從建立一個可運作的單一執行緒網頁伺服器作為起始。在我們開始之前，讓我們快速瞭解一下建構網頁伺服器會涉及到的協定。這些協定的細節超出本書的範疇，不過以下簡短的概覽可以提供你所需要知道的訊息。
 
-網頁伺服器會涉及到的兩大協定為**超文本傳輸協定（Hypertext Transfer Protocol, HTTP）**與**傳輸控制協定（Transmission Control Protocol, TCP）**。這兩種協定都是**請求-回應（request-response）**協定，這意味著**客戶端**會初始一個請求，然後**伺服器**接聽請求並提供回應給客戶端。協定會定義這些請求與回應的內容。
+網頁伺服器會涉及到的兩大協定為**超文本傳輸協定（Hypertext Transfer Protocol, HTTP）**與**傳輸控制協定（Transmission Control Protocol, TCP）**。這兩種協定都是**請求 - 回應（request-response）**協定，這意味著**客戶端**會初始一個請求，然後**伺服器**接聽請求並提供回應給客戶端。協定會定義這些請求與回應的內容。
```
Comment (weihanglo) [inline src/ch20-01-single-threaded.md]: I am not sure about when to use which dash. Found [this article](https://english.stackexchange.com/questions/28601/appropriate-dash-to-use-when-attributing-a-quotation/59320) though. Could you explain why this change?

File: style-guide.md [hunk 6]
```diff
@@ -5,7 +5,7 @@
 * Prefer title case for chapter/section headings, ex: `## Generating a Secret
   Number` rather than `## Generating a secret number`.
 * Prefer italics over single quotes when calling out a term, ex: `is an
-  *associated function* of` rather than `is an ‘associated function’ of`.
+  *associated function* of` rather than `is an 'associated function' of`.
```
Comment (weihanglo) [inline style-guide.md]: False positive.

File: tools/src/bin/link2print.rs [hunk 7]
```diff
@@ -199,8 +199,8 @@ more text"
     }
     #[test]
     fn parses_name_with_utf8() {
-        let source = r###"[user’s forum](the user’s forum)"###.to_string();
-        let target = r###"user’s forum at *the user’s forum*"###.to_string();
+        let source = r###"[user's forum](the user's forum)"###.to_string();
+        let target = r###"user's forum at *the user's forum*"###.to_string();
```
Comment (weihanglo) [inline tools/src/bin/link2print.rs]: False positive.

File: .github/workflows/autocorrect.yml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+name: AutoCorrect
```
Comment (weihanglo) [inline .github/workflows/autocorrect.yml]: Personally, I'd rather do it manually from time to time. Not sure about others opinions.

File: listings/ch13-functional-features/listing-13-05/output.txt [hunk 9]
```diff
@@ -2,5 +2,5 @@ $ cargo run
    Compiling closure-example v0.1.0 (file:///projects/closure-example)
     Finished dev [unoptimized + debuginfo] target(s) in 0.43s
      Running `target/debug/closure-example`
-呼叫閉包前[1, 2, 3]
+呼叫閉包前 [1, 2, 3]
```
Comment (weihanglo) [inline listings/ch13-functional-features/listing-13-05/output.txt]: Should either update the corresponding code example. And I also prefer a full-width colon instead of a space.

File: src/ch08-03-hash-maps.md [hunk 10]
```diff
@@ -41,8 +41,8 @@
 此程式會以任意順序印出每個配對：
 
 ```text
-黃隊: 50
-藍隊: 10
+黃隊：50
+藍隊：10
```
Comment (weihanglo) [inline src/ch08-03-hash-maps.md]: Corresponding code examples need updates as well.

**Review by weihanglo — COMMENTED** [review]
Thanks for the patch

File: src/ch08-03-hash-maps.md [hunk 11]
```diff
@@ -40,7 +40,7 @@
 
 此程式會以任意順序印出每個配對：
 
-```text
+```
```
Comment (weihanglo) [inline src/ch08-03-hash-maps.md]: Any reason for this change?

**Review by wusyong — APPROVED** [review]

File: src/ch08-03-hash-maps.md (hunk 11 above)
Comment (huacnlee) [reply src/ch08-03-hash-maps.md]: That is same

Comment (weihanglo) [reply src/ch08-03-hash-maps.md]: Sorry buddy I don't get it. Same for that?

The original TRPL [has that `text`](https://github.com/rust-lang/book/blob/2cd1b5593d26dc6a03c20f8619187ad4b2485552/src/ch08-03-hash-maps.md?plain=1#L73-L76) as well. It has a specific meaning in mdbook: unspecified code block is seen as Rust code. Thus `text` need to be added.

Comment (huacnlee) [reply src/ch08-03-hash-maps.md]: Oh, I mean code block in markdown, without language and `text` is same behavior.

I have forgot this is used in mdBook. Sorry.

I have reverted it.

Comment (weihanglo) [reply src/ch08-03-hash-maps.md]: Hmm... I might be wrong. That is the behaivor of rustdoc, but I am not sure whether mdbook does that. Anyway, we'd better align to what the original TRPL gives us. Thank you for your quick fix!

**Review by weihanglo — APPROVED** [review]
